CODINGSTYLE.md

```diff
@@ -0,0 +1,335 @@
+# Coding Style Guide
+
+The Scala.js project has pretty drastic rules in terms of coding style.
```

Comment:
try to make it about the users, not about what you care about, you want the project to welcome external contributions (at least I hope so?) as long as they adhere to these guidelines..
so instead of "it is our firm belief...", how about "this is one of the cornerstones that has enabled scala.js to maintain a clean...."